ksqldb-functional-tests/src/test/resources/query-validation-tests/multi-joins.json

```diff
@@ -0,0 +1,782 @@
+        {"topic": "OUTPUT", "key": 1, "value": {"S1_ID": 1, "T2_ID": 2, "T3_ID": 3}, "timestamp":  12}
+      ],
+      "post": {
+        "sources": [
+          {"name": "OUTPUT", "type": "stream", "schema": "K INT KEY, S1_ID BIGINT, T2_ID BIGINT, T3_ID BIGINT"}
+        ]
+      }
+    },
+    {
+      "name": "STT - II - rekey with expression",
+      "statements": [
+        "CREATE STREAM S1 (ROWKEY INT KEY, K INT, ID bigint) WITH (kafka_topic='left', value_format='JSON');",
+        "CREATE TABLE T2 (ROWKEY INT PRIMARY KEY, ID bigint) WITH (kafka_topic='right', value_format='JSON');",
+        "CREATE TABLE T3 (ROWKEY INT PRIMARY KEY, ID bigint) WITH (kafka_topic='right2', value_format='JSON');",
+        "CREATE STREAM OUTPUT as SELECT s1.ID, t2.ID, t3.ID FROM S1 JOIN T2 ON S1.K + 1 = T2.ROWKEY JOIN T3 ON S1.K + 1 = T3.ROWKEY;"
+      ],
+      "properties": {
+        "ksql.any.key.name.enabled": true
+      },
+      "inputs": [
+        {"topic": "right2", "key": 1, "value": {"id": 3}, "timestamp": 10},
+        {"topic": "right", "key": 1, "value": {"id": 2}, "timestamp": 11},
+        {"topic": "left", "key": 0, "value": {"k": 0, "id": 1}, "timestamp": 12}
+      ],
+      "outputs": [
+        {"topic": "_confluent-ksql-some.ksql.service.idquery_CSAS_OUTPUT_0-L_Join-repartition", "key": 1, "value": {"S1_K": 0, "S1_ID": 1, "S1_ROWTIME": 12, "S1_ROWKEY": 0, "S1_KSQL_COL_0": 1}, "timestamp": 12},
+        {"topic": "OUTPUT", "key": 1, "value": {"S1_ID": 1, "T2_ID": 2, "T3_ID": 3}, "timestamp":  12}
+      ],
+      "post": {
+        "sources": [
+          {"name": "OUTPUT", "type": "stream", "schema": "KSQL_COL_0 INT KEY, S1_ID BIGINT, T2_ID BIGINT, T3_ID BIGINT"}
+        ]
+      }
+    },
+    {
+      "name": "STT - II - select * ",
+      "statements": [
+        "CREATE STREAM S1 (ROWKEY INT KEY, ID bigint) WITH (kafka_topic='left', value_format='JSON');",
+        "CREATE TABLE T2 (ROWKEY INT PRIMARY KEY, ID bigint) WITH (kafka_topic='right', value_format='JSON');",
+        "CREATE TABLE T3 (ROWKEY INT PRIMARY KEY, ID bigint) WITH (kafka_topic='right2', value_format='JSON');",
+        "CREATE STREAM OUTPUT as SELECT * FROM S1 JOIN T2 ON S1.ROWKEY = T2.ROWKEY JOIN T3 ON S1.ROWKEY = T3.ROWKEY;"
+      ],
+      "properties": {
+        "ksql.any.key.name.enabled": true
+      },
+      "inputs": [
+        {"topic": "right2", "key": 0, "value": {"id": 3}, "timestamp": 10},
+        {"topic": "right", "key": 0, "value": {"id": 2}, "timestamp": 11},
+        {"topic": "left", "key": 0, "value": {"id": 1}, "timestamp": 12}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "key": 0, "value": {"S1_ROWTIME": 12, "S1_ROWKEY": 0, "S1_ID": 1, "T2_ROWTIME": 12, "T2_ROWKEY": 0, "T2_ID": 2, "T3_ROWTIME": 12, "T3_ROWKEY": 0, "T3_ID": 3}, "timestamp":  12}
```

Comment:
Boo: output should not include key columns  in the value :(

Reply:
should be transparently resolved with https://github.com/confluentinc/ksql/issues/3731